src/web/external/mongoose/mongoose.h

```diff
@@ -0,0 +1,6221 @@
+#define MG_ENABLE_SSL 1
+#define MG_ENABLE_THREADS 0
+#define MG_SSL_IF MG_SSL_IF_MBEDTLS
+// We provide a better mg_ssl_if_mbed_random method (than the dummy
+// used if not provided, as recommend).  See
+// src/services/web_service.cpp.
+//#define MG_SSL_MBED_DUMMY_RANDOM
+
+#ifdef MG_MODULE_LINES
+#line 1 "mongoose/src/common.h"
+#endif
+/*
+ * Copyright (c) 2004-2013 Sergey Lyubka
+ * Copyright (c) 2013-2015 Cesanta Software Limited
+ * All rights reserved
+ *
+ * This software is dual-licensed: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
```

Comment:
We need to determine if this is compatible with our Affero GPLv3 license before merging.

Reply:
Do we have a resident license expert?  From my look at it now, it looks incompatible given that it's licensed as GPLv2 and not "GPLv2 or later".

https://www.gnu.org/licenses/gpl-faq.en.html#AllCompatibility

"Each place that the matrix states GPLv3, the same statement about compatibility is true for AGPLv3 as well."

Reply:
No experts here. If it's incompatible we need to either figure out how to manage compliance or factor it out into an external lib.